metricbeat/module/kubernetes/event/event.go

```diff
@@ -114,7 +115,7 @@ func generateMapStrFromEvent(eve *kubernetes.Event) common.MapStr {
 	if len(eve.Metadata.Annotations) != 0 {
 		annotations := make(common.MapStr, len(eve.Metadata.Annotations))
 		for k, v := range eve.Metadata.Annotations {
-			annotations[common.DeDot(k)] = v
+			safemapstr.Put(annotations, k, v)
```

Comment:
I wonder if we should document this under breaking changes? As in case someone filtered for the annotations this queries now stop working. Will still merge the PR but we can discuss this as a follow up.

Reply:
Note: I agree this is a bug fix in general.

Reply:
Yep, I wonder how we should communicate this, so far without this change the only way to get some field that is renamed by this was using `drop_field` processor to ensure there are no mapping issues.

Reply:
As we link to the PR in the changelog could you add more details to the PR message? So people ending up on this PR get some background info?

Reply:
good point, done!